### ISSUE TITLE
PLW0602 false positive on list.append

### ISSUE THREAD

**Issue by AaronDMarasco** [astral-sh/ruff#10914]
It seems to only look for things like `=` or `+=` and doesn't recognize `list.append`.

* A minimal code snippet that reproduces the bug.

```python
probs = []
def y():
  global probs
  probs.append("problem")
```

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.

```sh
$ ruff check --isolated --config "lint.select=['PLW0602']" PLW0602.py
PLW0602.py:3:10: PLW0602 Using global for `probs` but no assignment is done
Found 1 error.
```

* The current Ruff settings (any relevant sections from your `pyproject.toml`).
  * N/A
  
* The current Ruff version (`ruff --version`).
  * ruff 0.3.7

**Comment by AaronDMarasco** [comment]
Never mind... because it's not rebinding the variable name and using the reference, it's cromulent...
